index.js

```diff
@@ -72,131 +72,233 @@ function isTriggerValid(triggerMode, delay) {
+    return (
+      typeof delay !== undefined &&
+      !isNaN(delay) &&
+      delay >= DELAY_MIN &&
+      delay <= DELAY_MAX
+    )
+  } else if (triggerMode === "FN_TRIGGER") {
     // fn trigger doesn't need any param
     return true
   }
 }
 
-const sleep = (time) => new Promise(resolve => {
-  setTimeout(resolve, time)
-})
+const sleep = time =>
+  new Promise(resolve => {
+    setTimeout(resolve, time)
+  })
 
 /**
  * Depending on the trigger mode, will wait for the trigger to occur and will
  * then resolve. In any case, the trigger is raced by a sleep on the MAX_DELAY
  * (either implicit or actual race)
  */
-const waitPreview = (triggerMode, page, delay) => new Promise(async (resolve) => {
-  let resolved = false
-  if (triggerMode === "DELAY") {
-    await sleep(delay)
-    resolve()
-  }
-  else if (triggerMode === "FN_TRIGGER") {
-    Promise.race([
-      // add event listener and wait for event to fire before returning
-      page.evaluate(function () {
-        return new Promise(function (resolve, reject) {
-          window.addEventListener("fxhash-preview", function () {
-            resolve() // resolves when the event fires
+const waitPreview = (triggerMode, page, delay) =>
+  new Promise(async resolve => {
+    if (triggerMode === "DELAY") {
+      await sleep(delay)
+      resolve()
+    } else if (triggerMode === "FN_TRIGGER") {
+      Promise.race([
+        // add event listener and wait for event to fire before returning
+        page.evaluate(function () {
+          return new Promise(function (resolve, reject) {
+            window.addEventListener("fxhash-preview", function () {
+              resolve() // resolves when the event fires
+            })
           })
-        })
-      }),
-      sleep(DELAY_MAX)
-    ]).then(resolve)
-  }
-})
+        }),
+        sleep(DELAY_MAX),
+      ]).then(resolve)
+    }
+  })
 
 // process the raw features extracted into attributes
 function processRawTokenFeatures(rawFeatures) {
   const features = []
   // first check if features are an object
-  if (typeof rawFeatures !== "object" || Array.isArray(rawFeatures) || !rawFeatures) {
+  if (
+    typeof rawFeatures !== "object" ||
+    Array.isArray(rawFeatures) ||
+    !rawFeatures
+  ) {
     throw new Error("Invalid features")
   }
   // go through each property and process it
   for (const name in rawFeatures) {
     // chack if propery is accepted type
-    if (!(typeof rawFeatures[name] === "boolean" || typeof rawFeatures[name] === "string" || typeof rawFeatures[name] === "number")) {
+    if (
+      !(
+        typeof rawFeatures[name] === "boolean" ||
+        typeof rawFeatures[name] === "string" ||
+        typeof rawFeatures[name] === "number"
+      )
+    ) {
       continue
     }
     // all good, the feature can be added safely
     features.push({
       name,
-      value: rawFeatures[name]
+      value: rawFeatures[name],
     })
   }
   return features
 }
 
+const extractFeatures = async page => {
```

Comment:
these utils are just extracted wholesale from the body of the handler to make it more readable